PostHog/PostHogSDK.swift

```diff
@@ -232,18 +234,39 @@ let maxRetryDelay = 30.0
         }
     }
 
-    @objc public func unregister(_ key: String) {
+    @objc(unregisterProperty:)
+    public func unregister(_ key: String) {
         personPropsLock.withLock {
             var props = getRegisteredProperties()
             props.removeValue(forKey: key)
             storage?.setDictionary(forKey: .registerProperties, contents: props)
         }
     }
 
-    @objc public func identify(_ distinctId: String,
-                               properties _: [String: Any]? = nil,
-                               userProperties: [String: Any]? = nil,
-                               userPropertiesSetOnce: [String: Any]? = nil)
+    @objc public func identify(_ distinctId: String) {
+        identify(distinctId, properties: nil, userProperties: nil, userPropertiesSetOnce: nil)
+    }
+
+    @objc(identifyWithDistinctId:properties:)
+    public func identify(_ distinctId: String,
+                         properties: [String: Any]? = nil)
+    {
+        identify(distinctId, properties: properties, userProperties: nil, userPropertiesSetOnce: nil)
+    }
+
+    @objc(identifyWithDistinctId:properties:userProperties:)
+    public func identify(_ distinctId: String,
+                         properties: [String: Any]? = nil,
+                         userProperties: [String: Any]? = nil)
+    {
+        identify(distinctId, properties: properties, userProperties: userProperties, userPropertiesSetOnce: nil)
+    }
+
+    @objc(identifyWithDistinctId:properties:userProperties:userPropertiesSetOnce:)
+    public func identify(_ distinctId: String,
+                         properties _: [String: Any]? = nil,
+                         userProperties: [String: Any]? = nil,
+                         userPropertiesSetOnce: [String: Any]? = nil)
```

Comment:
Typically in every other SDK we don't allow you to modify the normal properties of the `$identify` event. Mostly because it has special behaviour. I would remove it here, or at the very least make it the last argument in the chain, not the first.

Reply:
Mmm, I've taken this from https://github.com/PostHog/posthog-js-lite/blob/9ac7090d676dc2a6e5688cbe2b0ec001e2a23088/posthog-core/src/index.ts#L149 which allow you passing the properties

> Mostly because it has special behaviour

I don't have context about that, so I don't know what the right call here is, happy to remove/reorder.
JS SDK does not have it, should I remove it then since JS is the single source of truth?